### PR TITLE
Update Fortran interfaces in Struct

### PR DESCRIPTION
**Pull request by victorapm**
In this PR, the `HYPRE_*SetPrecond` functions implemented in `struct_ls/F90_HYPRE_struct_*.c` are updated so that all solvers support SMG, PFMG, SparseMSG, Jacobi and Diagonal Scaling as possible choices for the preconditioner. Previously, this support was incomplete. This is motivated by issue 1731 from hypre Issue Tracker. 

PS: should we keep supporting Jacobi and Diagonal Scaling in Struct? The latter can be used via the former one.

### CODE REVIEW

File: src/struct_ls/F90_HYPRE_struct_bicgstab.c
Comment (rfalgout) [inline src/struct_ls/F90_HYPRE_struct_bicgstab.c]: Hi @victorapm .  We are not advertising the SparseMSG solver (we probably should delete it actually, but let's not do that quite yet), so I would not add this to the Fortran files.  If they already appear somewhere, I would go ahead and remove them instead.

**Review by rfalgout — APPROVED** [review]
Hi @victorapm .  This looks good, except for the SparseMSG solver stuff.  Thanks for updating this!

File: src/struct_ls/F90_HYPRE_struct_bicgstab.c
Comment (victorapm) [reply src/struct_ls/F90_HYPRE_struct_bicgstab.c]: Ok, Rob! I'm fixing this with commit 24fb670

**Comment by victorapm** [comment]
Thanks, Rob! I removed support to SparseMSG in commit 24fb670. Can I go ahead and merge?

**Comment by rfalgout** [comment]
It’s fine with me.  Did you want @ulrikeyang to approve too?  I see that she is listed as a reviewer.

**Comment by victorapm** [comment]
Yes, Rob! I'll wait for her approval. Thanks!

**Review by ulrikeyang — APPROVED** [review]

**Pull request merged**
